### ISSUE TITLE
Improve the classical declarations and assignments

### ISSUE THREAD

**Issue by TheGupta2012** [qBraid/qbraid-qir#90]
### Feature Description

Continuation of #85. 

### Details 
Some tasks were left in the issue - 
1. **Array Initialization using** `ArrayLiteral`. Eg- 
```c
// 1-D 
array[int[32], 3] my_arr = {1, 2, 3};
```
This should be able to handle arbitrary dimensions eg. - 
```c
// N-D should also be supported
array[int[32], 2, 3] my_arr = {{1, 2, 3}, {4, 5, 6};
```

To be updated -
-  https://github.com/qBraid/qbraid-qir/blob/975744c4c740f36693e0bf9199015b7b82e82fad/qbraid_qir/qasm3/visitor.py#L841
 
2. **Type casting during assignment and expression evaluation**: Users should be able to cast types allowed in openqasm3. Reference - [type casting in `openqasm`](https://openqasm.com/language/types.html#casting-specifics)

To be updated - 
- Some logic change for -  https://github.com/qBraid/qbraid-qir/blob/975744c4c740f36693e0bf9199015b7b82e82fad/qbraid_qir/qasm3/visitor.py#L946 to handle "castable" values. Eg. An `int[32]` can be cast to `float[32]` but not vice versa.

3. **Validating constant value expressions to only use const variables**: Constant expression can not contain variables, it should be validated that the init expression only uses literals or other const variables. Reference - [constants in `openqasm`](https://openqasm.com/language/types.html#compile-time-constants).

To be updated - 
- It would be nice to have a flag in [self._evaluate_expression](https://github.com/qBraid/qbraid-qir/blob/975744c4c740f36693e0bf9199015b7b82e82fad/qbraid_qir/qasm3/visitor.py#L957) for evaluating the constant initial expressions. This could be set to `True` and the values coming in should then be validated whether they are constant or not.
- Another approach could be creating a new constant expression evaluator but may result in redundancy.

4. **Add complex type support** (optional)

5. **Add angle type support** (optional)

**Comment by nathanshammah** [comment]
Is this closed by #85?

**Comment by nathanshammah** [comment]
Got it, thanks!
